pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -74,16 +78,82 @@ func (s *statesInformer) syncNodeResourceTopology(node *corev1.Node) {
 	}
 }
 
+func (s *statesInformer) calGuaranteedCpu(usedCPUs map[int32]*extension.CPUInfo, stateJson string) ([]extension.PodCPUAlloc, error) {
+	if stateJson == "" {
+		return nil, fmt.Errorf("empty state file")
+	}
+	checkPoint := &state.CPUManagerCheckpoint{}
+	err := json.Unmarshal([]byte(stateJson), checkPoint)
+	if err != nil {
+		return nil, err
+	}
+	podAllocs := []extension.PodCPUAlloc{}
+	for pod := range checkPoint.Entries {
+		cpuSet := cpuset.NewCPUSet()
+		for container, cpuString := range checkPoint.Entries[pod] {
+			if tmpContainerCPUSet, err := cpuset.Parse(cpuString); err != nil {
+				klog.Errorf("could not parse cpuset %q for container %q in pod %q: %v", cpuString, container, pod, err)
+				continue
+			} else {
+				cpuSet = cpuSet.Union(tmpContainerCPUSet)
+			}
+		}
+		podAllocs = append(podAllocs, extension.PodCPUAlloc{
+			Name:   pod,
+			CPUSet: cpuSet.String(),
+		})
+		for _, cpuID := range cpuSet.ToSliceNoSort() {
+			delete(usedCPUs, int32(cpuID))
+		}
+	}
+	return podAllocs, nil
+}
+
 func (s *statesInformer) reportNodeTopology() {
+	klog.Info("start to report node topology")
 	s.nodeRWMutex.RLock()
 	nodeName := s.node.Name
 	s.nodeRWMutex.RUnlock()
 	ctx := context.TODO()
-
 	cpuTopology, usedCPUs, err := s.calCpuTopology()
 	if err != nil {
 		return
 	}
+	// TODO: support https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
+	cpuPolicy, stateFilePath, cpuManagerOpt, err := system.GuessCPUManagerOptFromKubelet()
+	if err != nil {
+		klog.Errorf("failed to guess kubelet cpu manager opt, err: %v", err)
+	}
+	cpuManagerPolicy := extension.CPUManagerPolicy{
+		Policy:  cpuPolicy,
+		Options: cpuManagerOpt,
+	}
+	cpuManagerPolicyJson, err := json.Marshal(cpuManagerPolicy)
+	if err != nil {
+		klog.Errorf("failed to marshal cpu manager policy, err: %v", err)
+		return
```

Comment:
consider remove the `return` because other fields are still need report, so as the following `return`.

Reply:
ok